File: SOURCES/app-tim-evxhw/pyapi/test-interruptions/provaints.py



import mevr
import signal
import time
import os
import fcntl

p = 0
er = mevr.new('/dev/evs/r13')


def handler(a, b):
	global p, er
	print '**** que passa neng\n'
	p = p+1
	er.SetVal(0x8, 0xff);
	er.IrqHandled()

signal.signal(signal.SIGIO, handler)

er = mevr.new('/dev/evs/r13')
#er = mevr.new('/tmp/mentida.bin')
# er.SetVal(0xc,0xff)

er.Enable(True)

me = (-1 << mevr.IRQ_MASTER_ENABLE)
hb = (1 << mevr.IRQFLAG_HEARTBEAT)
print int(me | hb)
er.IrqEnable(me | hb)


signal.signal(signal.SIGIO, handler)

er.IrqAssignHandler()


for i in range(0,5):
	print '-----------------------'
	er.DumpStatus()
	time.sleep(3)
	er.DumpStatus()

er.Enable(False)
er.Close()

print p

File: SOURCES/app-tim-evxhw/pyapi/test-interruptions/provaints.c

#include <stdlib.h>
#include <stdio.h>
// #include "erapi.h"

#include "erapi.c"

void handler(int n)
{
	printf("ei que passa neeeng %d\n", n);
}

int main()
{
	int fd;
	int oflags;

	struct MrfErRegs *pEr;

	fd = EvrOpen(&pEr, "/dev/evs/r13");
	oflags = fcntl(fd, F_GETFL);
	printf("PREFLAGS: %x\n", oflags);

	EvrEnable(pEr, 1);
		
// 	EvrSetVal(pEr, 0xc, 0xff);
	printf("0x%x\n", EvrIrqEnable(pEr, EVR_IRQ_MASTER_ENABLE | EVR_IRQFLAG_HEARTBEAT));
	EvrIrqAssignHandler(pEr, fd, handler);
	oflags = fcntl(fd, F_GETFL);
	printf("POSTFLAGS: %x\n", oflags);
	EvrSetVal(pEr, 0x8, 0xff);
	EvrDumpStatus(pEr);
	sleep(3);
	EvrDumpStatus(pEr);

// 	EvrEnable(pEr, 0);
	EvrClose(fd);
}

